Source/Dafny/RefinementTransformer.cs

```diff
@@ -269,6 +271,19 @@ public RefinementTransformer(Program p)
       }
     }
 
+    /// <summary>
+    /// Give unresolved newtypes a reasonable default type (<c>int</c>), to avoid having to support `null` in the
+    /// rest of the resolution pipeline.
+    /// </summary>
+    private void AddDefaultBaseTypeToUnresolvedNewtypes(List<TopLevelDecl> topLevelDecls) {
+      foreach (var d in topLevelDecls) {
+        if (d is NewtypeDecl { IsRefining: true, BaseType: null } decl) {
+          Reporter.Info(MessageSource.RefinementTransformer, decl.tok, $"defaulting to 'int' for unspecified base type of '{decl.Name}'");
```

Comment:
This happens only when the `newtype T ...` doesn't actually refine a `newtype` in a parent module, right? (Otherwise, this info message would come out for all program that refine a `newtype` and that would seem confusing to a user.)

Reply:
Correct, `BaseType == null` captures that